Kernel/FileSystem/InodeWatcher.h

```diff
@@ -8,37 +9,74 @@
 
 #include <AK/Badge.h>
 #include <AK/CircularQueue.h>
-#include <AK/WeakPtr.h>
+#include <AK/HashMap.h>
+#include <AK/NonnullRefPtr.h>
 #include <Kernel/API/InodeWatcherEvent.h>
 #include <Kernel/FileSystem/File.h>
-#include <Kernel/Lock.h>
 
 namespace Kernel {
 
 class Inode;
 
+// A specific description of a watch.
+struct WatchDescription : public RefCounted<WatchDescription> {
```

Comment:
This struct doesn't seem like it needs to be reference counted.

The "two" owners are both `InodeWatcher`, it's just spread across two `HashMap`s. We could reorganize it so that one of them is a `HashMap` with a `NonnullOwnPtr` value type, and the other is a HashMap with `WatchDescription*`